scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,374 @@
+    }
+
+    private void downloadDirectory(FTPClient ftpClient, String parentDir, String currentDir, String saveDir)
+            throws IOException {
+        String dirToList = parentDir;
+        if (!currentDir.equals("")) {
+            dirToList = Paths.get(dirToList, currentDir).toString();
+        }
+        FTPFile[] subFiles = ftpClient.listFiles(dirToList);
+        if (subFiles != null && subFiles.length > 0) {
+            for (FTPFile aFile : subFiles) {
+                String currentFileName = aFile.getName();
+                if (currentFileName.equals(".") || currentFileName.equals("..")) {
+                    // skip parent directory and the directory itself
+                    continue;
+                }
+                String filePath = Paths.get(parentDir, currentDir, currentFileName).toString();
+                if (currentDir.equals("")) {
+                    filePath = Paths.get(parentDir, currentFileName).toString();
+                }
+                String newDirPath = Paths.get(saveDir, parentDir, currentDir, currentFileName).toString();
+                if (currentDir.equals("")) {
+                    newDirPath = Paths.get(saveDir, parentDir, currentFileName).toString();
+                }
+
+                if (aFile.isDirectory()) {
+                    // create the directory in saveDir
+                    File newDir = new File(newDirPath);
+                    boolean created = newDir.mkdirs();
+                    if (created) {
+                        getOut().println("CREATED the directory: " + newDirPath);
+                    } else {
+                        getOut().println("COULD NOT create the directory: " + newDirPath);
+                    }
+
+                    // download the sub directory
+                    downloadDirectory(ftpClient, dirToList, currentFileName, saveDir);
+                } else {
+                    // download the file
+                    boolean success = downloadSingleFile(ftpClient, filePath, newDirPath);
+                    if (success) {
+                        getOut().println("DOWNLOADED the file: " + filePath);
+                    } else {
+                        getOut().println("COULD NOT download the file: " + filePath);
+                    }
+                }
+            }
+        }
+    }
+
+    private boolean downloadSingleFile(FTPClient ftpClient, String remoteFilePath, String savePath) throws IOException {
+        getOut().println("About to download the file: " + remoteFilePath);
+        File downloadFile = new File(savePath);
+
+        File parentDir = downloadFile.getParentFile();
+        if (parentDir != null && !parentDir.exists()) {
+            parentDir.mkdirs();
+        }
+        OutputStream outputStream = new BufferedOutputStream(new FileOutputStream(downloadFile));
+
+        try {
+            return ftpClient.retrieveFile(remoteFilePath, outputStream);
+        } catch (IOException ex) {
+            throw ex;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add logic to this catch clause or eliminate it and rethrow the exception automatically. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2737)